src/Metadata/IgnoreDeprecations.php

```diff
@@ -9,15 +9,37 @@
  */
 namespace PHPUnit\Metadata;
 
+use function preg_match;
+
 /**
  * @immutable
  *
  * @no-named-arguments Parameter names are not covered by the backward compatibility promise for PHPUnit
  */
 final readonly class IgnoreDeprecations extends Metadata
 {
+    /** @var null|non-empty-string */
+    private ?string $messagePattern;
+
+    /**
+     * @param int<0, 1>             $level
+     * @param null|non-empty-string $messagePattern
+     */
+    protected function __construct(int $level, null|string $messagePattern)
+    {
+        parent::__construct($level);
+
+        $this->messagePattern = $messagePattern;
+    }
+
     public function isIgnoreDeprecations(): true
     {
         return true;
     }
+
+    public function shouldIgnore(string $message): bool
```

Comment:
I am not comfortable with the fact that a value object, which is what a `Metadata` object is supposed to be, makes a decision like that.

This decision should be moved to `ErrorHandler::deprecationIgnoredByTest()`, for example.